lib/bulkhead/Bulkhead.ts

```diff
@@ -0,0 +1,54 @@
+import { DEFAULT_SIZE } from './BulkheadOptions';
+
+export class Bulkhead {
+
+  private inFlight: number = 0;
+  private queue: any[] = [];
+
+  constructor(
+    private readonly threshold: number,
+    private readonly size: number = DEFAULT_SIZE,
+  ) { }
+
+  public pass(): boolean {
+    return !this.size || this.queue.length < this.size;
```

Comment:
What it `this.size === 0`? I think better is to check `this.size === undefined`.